BFS_Connected_Components/Q2_Monkey_Fruits.py

```diff
@@ -0,0 +1,60 @@
+import math
+from collections import deque
+
+
+def build_graph(trees):
+    adj_list = [[] for _ in range(len(trees))]
+
+    for tree1_id, tree1 in enumerate(trees):
+        for tree2_id, tree2 in enumerate(trees):
+            if tree1_id == tree2_id:continue
+
+            x_one, y_one, num_fruits_one, vine_length_one = tree1
+            x_two, y_two, num_fruits_two, vine_length_two = tree2
+            
+            distance = math.sqrt((x_one - x_two)**2 + (y_one - y_two)**2)
+            
+            if vine_length_one < distance or vine_length_two < distance: continue
```

Comment:
I think there is some human meaning that is not represented by the code here. Specifically, what we're checking here is whether or not the monkey can travel freely between these trees. I'd do something like this:
```
can_travel_freely_between_trees = vine_length_one >= distance and vine_length_two >= distance
if not can_travel_freely_between_trees: continue
```